### DISCUSSION TITLE
[Proposal] Follow SemVer in versioning. Offer free upgrades for 12 months instead of the current version.

### DISCUSSION THREAD

**Discussion by tabacitu** [Laravel-Backpack/community-forum#322] [General]
## Until Backpack 4.1

This is what `major.minor.patch` meant for Backpack:

- `major` - **PAID upgrade;** MAJOR breaking changes; historically every 2-3 years; upgrading may take even 2-3 hours; includes MAJOR new features, MAJOR changes in how the whole system works, and complete rewrites; it allows us to considerably improve how we build admin panels and add features that were previously impossible;
- `minor` - **FREE upgrade;** SOME breaking changes; historically every 6 months; upgrading takes 15-60 minutes; offers big new features, for free;
- `patch` - **FREE upgrade;** NO breaking changes; historically every week; upgrading can be done automatically with composer; includes bug fixes and non-breaking new features;

So it's basically `paid.upgrade.patch`.

PROs: 
- `paid.upgrade.patch` makes the second segment actually stand for something. You know by looking at the version name if it's a paid upgrade or not.

CONs:
- `paid.upgrade.patch` makes the free upgrades we push every 6 months seem less important (6 months of work on an new version is NOT minor after all); 
- `paid.upgrade.patch` probably affects developer perception, in terms of _how important_ it is to upgrade to the latest Backpack version;
- `paid.upgrade.patch` makes it more difficult to install Backpack (you have to know which version you want to install, because you have to do `composer require backpack/crud:"4.1.*"`);

## Starting with September 2020

I think it's a good idea that we start following SemVer:
- Backpack 4.2 will be named Backpack 5, even if it's a free upgrade;
- Backpack 4.3 will be named Backpack 6, even if it's a free upgrade (don't know yet);
- the next paid version will be called Backpack 7 (or whenever we decide to create a new paid version);

SemVer has a major downside - that your will no longer be able to tell from the version number if it's a paid upgrade or not. But I think that's solvable with good documentation, and a clear button in your Backpack Account that tells you if the upgrade is free or paid.

This is more of a rambling or note-to-self kind of issue, but I'm open to feedback if anybody has thoughts on this. There's a lot of time until 4.2 (Sep-Oct 2020) so I'm happy to change plans in a different direction if it proves not to be a good idea.

**Comment by jcastroa87** [comment]
Hi.

Due to not activity on many years i will close this discussion, if there is needed feel free to reopen or create a new one.

Cheers.
